velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -951,4 +951,53 @@ struct MillisToTimestampFunction {
   }
 };
 
+template <typename T>
+struct TimestampDiffFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const std::vector<TypePtr>& /*inputTypes*/,
+      const core::QueryConfig& config,
+      const arg_type<Varchar>* unitString,
+      const arg_type<Timestamp>* /*timestamp1*/,
+      const arg_type<Timestamp>* /*timestamp2*/) {
+    VELOX_USER_CHECK_NOT_NULL(unitString);
+    unit_ = fromDateTimeUnitString(
```

Comment:
Should we support the micro seconds in fromDateTimeUnitString? If that, this issue https://github.com/facebookincubator/velox/pull/13535/files#r2168040923 needs to be resolved

Reply:
No problem here, fromDateTimeUnitString supports micro seconds. Presto and spark are distinguished by the parameter allowMicro.